woocommerce/google-listings-and-ads
# Label campaigns for the web version and the WC Mobile app

### Changes proposed in this Pull Request:



Part of pcTzPl-2nS-p2

This PR will automatically label campaigns created via the WC Mobile App by checking the user agent headers. The labels used will be `wc-android` or `wc-ios`. If the campaign isn't created through the WC Mobile App, it will be labeled as `wc-web`.

### Screenshots:




### Detailed test instructions:


1. Download the WC Mobile App and connect it to one of your local sites. Then, navigate to WC Admin -> Google for WooCommerce and create a new campaign. Alternatively,  simulate this process by sending a request to `POST /wp-json/wc/gla/ads/campaigns` with the `User Agent` header set to something like: `Mozilla/5.0 (iPhone; CPU iPhone OS 17_5_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 wc-ios/19.7.1`. Set the below body:

```
{
    "amount": 0.01,
    "targeted_locations": ["ES"],
}
```

2. Follow the steps mentioned in this PR to check if the label has been set: https://github.com/woocommerce/google-listings-and-ads/pull/2485. It has to be `wc-ios` or `wc-android`.
3. Repeat the process on a laptop, and the campaign label should be `wc-web`


### Additional details:


### Changelog entry

> Tweak - Label campaigns for the web version and the WooCommerce Mobile app.
